### PR TITLE
Start using the best resilience expansion in SIL

### PR DESCRIPTION
**Pull request by slavapestov**
Follow-up to https://github.com/apple/swift/pull/23263.

Finally start using the best resilience expansion in SIL type lowering.

Completes rdar://problem/24057844, https://bugs.swift.org/browse/SR-261.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please benchmark

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

File: test/SILOptimizer/globalopt_resilience.swift [hunk 1]
```diff
@@ -21,15 +21,38 @@ public struct ResilientStruct {
...
+// CHECK: [[S2:%.*]] = struct $ResilientStruct ([[S1]] : $Int)
+// CHECK: [[TMP:%.*]] = alloc_stack $ResilientStruct
+// CHECK: store [[S2]] to [[TMP]] : $*ResilientStruct
+// CHECK: [[METHOD:%.*]] = function_ref @$s4test15ResilientStructV6methodyyF : $@convention(method) (@in_guaranteed ResilientStruct) -> ()
+// CHECK: apply [[METHOD]]([[TMP]]) : $@convention(method) (@in_guaranteed ResilientStruct) -> ()
+// CHECK: [[RESULT:%.*]] = tuple ()
+// CHECK: return [[RESULT]] : $()
+
+// CHECK-LABEL: sil @$s4test26cannotConvertToValueUseAltyyF : $@convention(thin) () -> ()
+// CHECK: [[TMP:%.*]] = alloc_stack $ResilientStruct
```
Comment (atrick) [inline test/SILOptimizer/globalopt_resilience.swift]: I would like at least a FIXME here.

FIXME: The optimizer should easily eliminate this copy. The problem is that copy forwarding assumes that address-only types are always explicitly destroyed.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Pull request merged**